packages/commonwealth/server/migrations/20230530140014-undo-role-migration.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    await queryInterface.sequelize.transaction(async (t) => {
+      await queryInterface.addColumn(
+        'Addresses',
+        'is_user_default',
+        {
+          type: Sequelize.BOOLEAN,
+          allowNull: false,
+          defaultValue: false,
+        },
+        { transaction: t }
+      );
+      await queryInterface.addColumn(
+        'Addresses',
+        'role',
+        {
+          type: Sequelize.ENUM('member', 'moderator', 'admin'),
+          allowNull: false,
+          defaultValue: 'member',
+        },
+        { transaction: t }
+      );
+
+      const roles = await queryInterface.sequelize.query(
+        `SELECT address_id, permission FROM "Roles"`,
+        { type: queryInterface.sequelize.QueryTypes.SELECT, transaction: t }
+      );
+
+      const ids = roles.map((r) => r.address_id);
+      const permissions = roles.map((r) => r.permission);
+
+      // Make the update in one query, cast the permissions to the enum
+      await queryInterface.sequelize.query(
+        `
+        UPDATE "Addresses"
+        SET role = CASE 
+            ${ids
+              .map(
+                (id, index) =>
+                  `WHEN id = ${id} THEN '${permissions[index]}'::"enum_Addresses_role"`
+              )
+              .join(' ')}
+        END
+        WHERE id IN (${ids.join(', ')})
+      `,
+        { transaction: t }
+      );
+
+      // query chain to a string containing allowed roles, example 1inch -> admin, moderator, member
```

Comment:
We can guarantee that admin, moderator, and member are the only roles allowed. We never created more. No need to account for the possibility.

Reply:
Maybe a micro optimization but I think it is more efficient if we combine these columns into 1 instead of having 3 separate boolean columns like (allow_member, allow_moderator, allow_admin). It also has the benefit of grouping these together so that when we query for it the user knows that these fields are related. If you think this way is better, I still am fine with going with the 3 boolean approach, but regardless @NakulManchanda If you can, i'd like a 3rd opinion here.

Reply:
The allow columns are for the way that which roles in the community were "allowed" to be made. In the implementation we had, each community had communityRoles entries. Each row represented whether a role was allowed. If for example, a community did not have the moderator role, then it would not be able to be created in the createAddress endpoint.

Reply:
Resolved via slack